### DISCUSSION TITLE
Different Conditional WGANGP performances

### DISCUSSION THREAD

**Discussion by ncuxomun** [Lightning-AI/pytorch-lightning#7763] [General]
Hi, guys!

I am running Conditional WGANGP (same file) on:
1. My Windows machine on NVIDIA 2070 Max-Q
2. Another Windows CPU machine
3. Google colab on a GPU (the free one)

All three machines/runs are so different in terms of loss values and outcome qualities of the generated images. For example, at Epoch 200,  the loss values are vastly different between machine 1 and 2. I understand there are hardware differences, but how can it be so different?

Thank you for your attention!

**Comment by justusschock** [comment]
Hi, 
Did you provide any seeds? Can you share your code?

**Reply by ncuxomun** [reply]
Hi, thanks for responding! 

Please follow this link to see the code:
https://github.com/PyTorchLightning/pytorch-lightning/discussions/7754

As for seeding and reproducibility, I defined the following:

manualSeed = 999
print('Random Seed:', manualSeed)
random.seed(manualSeed)
torch.manual_seed(manualSeed)
pl.seed_everything(manualSeed)
torch.cuda.manual_seed(manualSeed)
torch.backends.cudnn.benchmark = False
torch.backends.cudnn.determinstic = True

**Reply by justusschock** [reply]
This looks good to me. If you run the model on the same machine two times, do they produce the same results?

**Reply by ncuxomun** [reply]
Yes, indeed, I get the same results if I run the model on the same machine two or more times.
